### PR TITLE
269 parameter checking

### LINKED ISSUE #269: Error checking for rest api query parameters
**Detailed Description**
Each parameter in the rest api is not validated explicitly against the acceptable ranges/lists.

**Expected Behavior**
To avoid errors or partial processing, each query parameter should be validated before processing the query

**Environment:**
- Version: [e.g. 0.2.0-rc3]

**Additional Context**
If not checked, these parameters could hit the SQL query and cause errors in the log file.

### PR DESCRIPTION
**Pull request by atul-hedera**
**Detailed description**:
Added checks for query parameters.

**Which issue(s) this PR fixes**:
Fixes #269 

**Checklist**
- [ ] Documentation added
- [x] Tests updated - added negative tests

### CODE REVIEW

File: rest-api/utils.js [hunk 1]
```diff
@@ -31,6 +37,98 @@ function isNumeric(n) {
...
+ * @return {Boolean} true if the parameter is valid. false otherwise
+ */
+const paramValidityChecks = function (param, val) {
+    let ret = false;
+
+    if (val === undefined) {
+        return (ret);
+    }
+
+    val = val.replace(/(gte?\:|lte?\:|eq\:)/, '');
```
Comment (mike-burrage-hedera) [inline rest-api/utils.js]: This should be the exact same function that the parser uses.
There should be 1 single "split value to operator and value" function.
As is - this specific one doesn't cover other operators ("ne" is in parseComparatorSymbol) and doesn't anchor the regex "account.id=0.0.eq:2" would be ok.

File: rest-api/__tests__/testutils.js [hunk 2]
```diff
@@ -123,7 +123,31 @@ const parseSqlQueryAndParams = function(sqlquery, sqlparams, orderprefix = '') {
     }
 }
 
+const testBadParams = function (request, server, url, param, badValues) {
+    const opList = ['', 'eq:', 'lt:', 'gt:', 'lte:', 'gte:'];
```
Comment (mike-burrage-hedera) [inline rest-api/__tests__/testutils.js]: "ne"?

File: rest-api/balances.js [hunk 3]
```diff
@@ -79,7 +86,7 @@ const getBalances = function (req) {
         sqlQuery += " join t_entities e\n" +
             " on e.entity_realm = ab.account_realm_num\n" +
             " and e.entity_num = ab.account_num\n" +
-            " and e.entity_shard = process.env.SHARD_NUM\n" +
+            " and e.entity_shard = " + process.env.SHARD_NUM + "\n" +
```
Comment (mike-burrage-hedera) [inline rest-api/balances.js]: You should rebase from master.

File: rest-api/utils.js [hunk 4]
```diff
@@ -31,6 +37,98 @@ function isNumeric(n) {
...
+
+    // Check the validity of every query parameter
+    for (const key in req.query) {
+        if (Array.isArray(req.query[key])) {
+            for (const val of req.query[key]) {
+                if (!paramValidityChecks(key, val)) {
+                    ret = {
+                        isValid: false,
+                        code: httpStatusCodes.BAD_REQUEST,
+                        contents: `Invalid parameter: ${key}`
```
Comment (mike-burrage-hedera) [inline rest-api/utils.js]: The successful response is in JSON, the error response should be as well.
{"_status": {"messages": [{"message": "Invalid parameter ${key}"]}} or something like that.

File: rest-api/balances.js (hunk 3 above)
Comment (atul-hedera) [reply rest-api/balances.js]: Done

File: rest-api/utils.js (hunk 4 above)
Comment (atul-hedera) [reply rest-api/utils.js]: Done

File: rest-api/balances.js (hunk 3 above)
Comment (atul-hedera) [reply rest-api/balances.js]: Done

File: rest-api/__tests__/testutils.js (hunk 2 above)
Comment (atul-hedera) [reply rest-api/__tests__/testutils.js]: Done

File: rest-api/utils.js (hunk 1 above)
Comment (atul-hedera) [reply rest-api/utils.js]: Added the split and checks for ne with the anchor. Eventually, I'd like to really use this exact function to return the <param, operator, value> tuple list from this function and simplify the parsing significantly.

**Pull request merged**